pkgs/stdenv/generic/setup.sh

```diff
@@ -120,15 +120,15 @@ exitHandler() {
     exitCode="$?"
     set +e
 
-    if [ -n "${showBuildStats:-}" ]; then
-        times > "$NIX_BUILD_TOP/.times"
-        local -a times=($(cat "$NIX_BUILD_TOP/.times"))
+    if [[ -n "${showBuildStats:-}" ]]; then
```

Comment:
Is there a reason for this line change? Sounds like `[` is enough here…

Reply:
This was more around trying to have a single way to do conditionals.
As I understand it, you never loose by using `[[` instead of `[` and there are less pitfalls.
I do understand that it might be controversial. If you prefer `[`, I'm happy to try to avoid that change where unecessary in future PRs.

Reply:
I doubt there is a coherent style guidance on that… this change was just a part of a kind of unrelated (except for adjacency) commit.

(I myself probably lean towards «POSIX except where extension make things much simpler», but I do not change `stdenv` much)